tests/student_comment_test.php

```diff
@@ -0,0 +1,206 @@
+<?php
```

Comment:
Great to see some unit tests in here.

I'd love to see some Behat test(s) as well - this new feature is very much a UI-focused update, so I'd expect a behat test that walked through a student adding a comment and confirmed that the teacher was able to view it.

There are lots of example behat tests already in the plugin, so it should be pretty straightforward to copy an existing test and modify it. Give me a shout if you need more guidance.

Reply:
Hi @davosmith , just wanted to let you know I'm still working on all your great feedback. 

Writing some behat tests is giving me grief though. I think the ideal would be to do something like 

```
And I set the field "comment1" to "something"
```
But I don't see how to set/access dynamic field names (as in there could be any number of comment input fields). 
I am also trying to use xpath with 

```
And I click on "(//span[@class='studentcomment']/input)[N]" "xpath_element"
And I type "this is my comment"
```
But it's pretty ugly. Do you know of a good example of using dynamic fields with behat?

Reply:
Some advice I remember hearing (I think it was from Tim Hunt) is that if you are struggling to write a good behat test, then you've probably got the accessibility on the element wrong.

My approach in this situation would be to add a label to the form element (hidden using the class 'accesshide') and make sure the label is something like "Comment on 'Watched intro video'" (or "Comment on checklist item 'Watched intro video'", if you want to be a bit more wordy). That way you can easily write:
```
And I set the following fields to these values:
| Comment on 'Watched intro video' | My comment |
```
(as an aside, I like that format for setting field values, even when only setting a single value, as I think it stands out and is easier to read when looking through the feature file).

Reply:
That actually helps a lot, yes that advice matches up with what I saw here https://docs.moodle.org/dev/Writing_acceptance_tests#Don.27t_use_XPath_or_CSS_selectors_-_fix_your_Accessibility_bugs
Thanks for the quick response! I'll have a go with that 👍

Reply:
Hi @davosmith , I had a lot of trouble but got one new behat test in that checks a comment is saved by a student and viewable by the teacher.